### ISSUE TITLE
Wrong ALPB energies with added salt.

### ISSUE THREAD

**Issue by nielskm** [grimme-lab/xtb#499]
**Describe the bug**
When running GFN2-xTB calculations with ALPB the energy does not converge to the zero-ion-strength limit.

**To Reproduce**
Try to run energy calculations with the Still kernel (the only one that supports adding salt) with `ion_str=0.0`, `ion_str=1.e-14`, `ion_str=1.e-12`, etc. You will see that the calculations with non-zero ion strength converge towards a limit that is significantly different from the result obtained with `ion_str=0.0`.

Attached below are a series of calculations on Na+, but the same applies to closed-shell organic molecules. I have used the latest master branch (9cc3c208a923f91fd662d132c810976fd089ef3b) built with ifort and linked with MKL.

[Na+.tar.gz](https://github.com/grimme-lab/xtb/files/6765619/Na%2B.tar.gz)

**Expected behaviour**
I would expect the energy to converge to the I=0 limit.

**Comment by awvwgk** [comment]
ALPB currently does not implement the salt screening terms, since they were not derived for the P16 interaction kernel if I recall correctly.

**Comment by nielskm** [comment]
It is correct that the salt screening is not implemented for the P16 kernel, which is why I have used the Still kernel in my tests (and also in the reference I=0 calculation).

**Comment by awvwgk** [comment]
Right, sorry. There should also be an additional contribution from the electrostatic shape term in ALPB with salt screening, which is not implemented yet. Given the current state of the salt screening implementation, it should only be available with GBSA and Still interaction kernel.

Adding the salt screening contribution for the electrostatic shape term in ALPB should be straight-forward enough, only the derivative might get a bit tedious. I'll mark this as a possible enhancement to the current solvation model.

**Comment by nielskm** [comment]
Thank you very much for the explanation @awvwgk !
